src/awkward/operations/structure.py

```diff
@@ -4177,13 +4175,9 @@ def with_cache(array, cache, highlevel=True, behavior=None):
 
     def getfunction(layout):
         if isinstance(layout, ak.layout.VirtualArray):
-            if cache is None:
-                newcache = layout.cache
-            elif layout.cache is None:
-                newcache = cache
             return lambda: ak.layout.VirtualArray(
                 layout.generator,
-                newcache,
+                cache,
```

Comment:
This change should also be reverted. `newcache` only exists to switch between `cache` and `layout.cache`. In fact, flake8 should have complained that `newcache` was being defined without being used.

I _think_ that the only change this PR needs is to disable the possibility of `highlevel=False`.

Reply:
I missed the fact that lines 4180 through 4183 are red (removed).